feat: accoil analytics ui

accoilmj opened the pull request:
## What are the changes introduced in this PR?

Adds new UI for 'Accoil Analytics' destination.

## Please explain the objectives of your changes below

Adds UI for Accoil Analytics.

Destination code PR: https://github.com/rudderlabs/rudder-transformer/pull/3873
Documentation PR: https://github.com/rudderlabs/rudderstack-docs/pull/1196

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new checks got introduced or modified in test suites. Please explain the changes.

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] I have executed schemaGenerator tests and updated schema if needed

- [x] Are sensitive fields marked as secret in definition config?

- [x] My test cases and placeholders use only masked/sample values for sensitive fields

- [x] Is the PR limited to 10 file changes & one task?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added configuration files for Accoil Analytics destination
	- Introduced comprehensive configuration and validation for analytics integration
	- Implemented consent management settings across multiple platforms

- **Documentation**
	- Added detailed UI configuration and schema for destination setup
	- Included test cases for configuration validation

- **Tests**
	- Created validation test suite for Accoil Analytics destination configurations
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

manish339k commented on src/configurations/destinations/accoil_analytics/db-config.json:
```diff
@@ -0,0 +1,121 @@
+{
+  "name": "Accoil_Analytics",
```
```suggestion
  "name": "ACCOIL_ANALYTICS",
```

manish339k commented on src/configurations/destinations/accoil_analytics/db-config.json:
```diff
@@ -0,0 +1,121 @@
+  "displayName": "Accoil Analytics",
+  "config": {
+    "cdkV2Enabled": true,
+    "transformAtV1": "processor",
```
```suggestion
    "transformAtV1": "router",
```

manish339k commented on src/configurations/destinations/accoil_analytics/db-config.json:
```diff
@@ -0,0 +1,121 @@
+      "warehouse": ["cloud"]
+    },
+    "destConfig": {
+      "defaultConfig": ["apiKey", "blacklistedEvents", "whitelistedEvents", "eventFilteringOption"],
```
why are we including other keys? I see only apiKey is in use. We should remove the other keys?

accoilmj replied:
Good catch. I've removed the others and left apiKey.

manish339k commented on src/configurations/destinations/accoil_analytics/ui-config.json:
```diff
@@ -0,0 +1,196 @@
+                    "configKey": "apiKey",
+                    "regex": "^[0-9a-zA-Z_]{1,80}$",
+                    "regexErrorMessage": "Invalid API Key",
+                    "placeholder": "Your Accoil.com API Key. You can find your API Key in your Accoil account settings.",
```
placeholder should be different, It should be something like e.g. 5fXXXX0. 
and add note like this `"note": "Your Accoil.com API Key. You can find your API Key in your Accoil account settings."`

accoilmj replied:
I've switched the placeholder to note and put an eg

manish339k commented on src/configurations/destinations/accoil_analytics/ui-config.json:
```diff
@@ -0,0 +1,196 @@
+                    "type": "textInput",
+                    "label": "API Key",
+                    "configKey": "apiKey",
+                    "regex": "^[0-9a-zA-Z_]{1,80}$",
```
this apiKey regex is not matching with apiKey property in schema.json

accoilmj replied:
I've re-generated schema.json which has picked up the changed regex.

accoilmj commented on src/configurations/destinations/accoil_analytics/schema.json:
```diff
@@ -0,0 +1,861 @@
+                  "default": "oneTrust"
+                },
+                "consents": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "consent": {
+                        "type": "string",
+                        "pattern": "(^\\{\\{.*\\|\\|(.*)\\}\\}$)|(^env[.].+)|^(.{0,100})$"
+                      }
+                    }
+                  }
+                }
+              },
+              "allOf": [
+                {
+                  "if": {
+                    "properties": {
+                      "provider": {
+                        "const": "custom"
+                      }
+                    },
+                    "required": ["provider"]
+                  },
+                  "then": {
+                    "properties": {
+                      "resolutionStrategy": {
+                        "type": "string",
+                        "enum": ["and", "or"]
+                      }
+                    },
+                    "required": ["resolutionStrategy"]
+                  }
+                }
+              ]
+            }
+          }
+        }
+      },
```
This file is autogenerated

accoilmj commented on src/configurations/destinations/accoil_analytics/schema.json:
```diff
@@ -0,0 +1,861 @@
+            }
+          },
+          "flutter": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "purpose": {
+                  "type": "string",
+                  "pattern": "(^\\{\\{.*\\|\\|(.*)\\}\\}$)|(^env[.].+)|^(.{0,100})$"
+                }
+              }
+            }
+          },
+          "cordova": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "purpose": {
+                  "type": "string",
+                  "pattern": "(^\\{\\{.*\\|\\|(.*)\\}\\}$)|(^env[.].+)|^(.{0,100})$"
+                }
+              }
+            }
+          },
+          "shopify": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "purpose": {
+                  "type": "string",
+                  "pattern": "(^\\{\\{.*\\|\\|(.*)\\}\\}$)|(^env[.].+)|^(.{0,100})$"
+                }
+              }
+            }
+          }
+        }
+      },
```
This file is autogenerated

devops-github-rudderstack commented:
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

accoilmj commented:
Rebased and pushed to fix staleness

manish339k commented:
Please add data validation test.
ref: https://github.com/rudderlabs/rudder-integrations-config/blob/develop/test/data/validation/destinations/active_campaign.json

accoilmj commented:
> Please add data validation test. ref: https://github.com/rudderlabs/rudder-integrations-config/blob/develop/test/data/validation/destinations/active_campaign.json

@manish339k I've added this test. When running locally it complained about having one trust cookie and ketch enabled. I removed them from the ui-config and now it passes. I'm not sure if this is correct though?

manish339k commented on src/configurations/destinations/ga4_v2/ui-config.json:
```diff
@@ -18,7 +18,9 @@
                     "type": "nestedSelect",
```
please exclude this file from PR.

accoilmj replied:
I've removed this. It seems to get generated each test run and I accidentally committed it this time.

manish339k commented:
cc: @utsabc / @ItsSudip for review.

devops-github-rudderstack commented:
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

manish339k commented:
merging this to the feature branch.
